src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -249,13 +262,15 @@ module.exports = function<T, P, I, TI, C>(
     }
     var fn = workInProgress.type;
     var props = workInProgress.pendingProps;
+    var context = getMaskedContext(workInProgress);
+
     var value;
 
     if (__DEV__) {
```

Comment:
I don't think there's any reason for this, putting just current owner assignment in a condition would be equivalent.